racket/collects/racket/private/for.rkt

```diff
@@ -418,6 +419,8 @@
                                               (make-rearm)))]
                     [else (eloop #f #'expanded-rhs)]))]
                [_ (eloop #f #'expanded-rhs)]))]
+          ;; when for:no-implicit-optimization is true
+          [[(id ...) rhs] (eloop #f #'rhs)]
```

Comment:
I guess another really minor issue is, the name `expanded-rhs` might not be appropriate anymore, since `rhs` here is not expanded.